test/tests/functional/pbs_preemption.py

```diff
@@ -733,3 +733,57 @@ def test_preempt_requeue_resc(self):
 
         self.server.expect(JOB, {'job_state': 'R'}, id=hjid)
         self.server.expect(JOB, {'job_state=Q': 1})
+
+    def test_preempt_wrong_cull(self):
+        """
+        Test to make sure that if a preemptor cannot run because
+        it misses a non-consumable on a node, preemption candidates
+        are not incorrectly removed from consideration
+        if and because they "do not request the relevant resource".
+        Deciding on their utility should be left to the check
+        to see whether the nodes they occupy are useful.
+        """
+        attr = {}
+        attr['type'] = 'string_array'
+        attr['flag'] = 'h'
```

Comment:
Consider putting this all on one line.

Reply:
Quite:

        attr = {'type': 'string_array', 'flag': 'h'}